ndb/src/google/cloud/ndb/_ports.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Ported implementations from the Google App Engine SDK.
+
+The following members have been brought in:
+
+* ``google.appengine.api.datastore_types.BlobKey``
+"""
+
+import functools
+
+from google.cloud.ndb import exceptions
+
+
+_MAX_STRING_LENGTH = 1500
+
+
+@functools.total_ordering
+class BlobKey:
```

Comment:
This is not a perfect implementation of [`datastore_types.BlobKey`][1]. A few differences:

- [`ValidateString()`][2] allows a 2.7 `basestring`, but I'm only allowing `bytes` (i.e. not `str` in Python 3 / `unicode` in Python 2)
- `ValidateString()` adds a special case for `datastore_types.Blob`, which is a thin wrapper over `bytes` (`str` in Python 2). We don't have this type and I don't plan on adding it

Also, ISTM that two things are strange:

- `None` shouldn't be allowed as a blob key
- `len(blob_key) > 1500` shouldn't matter, i.e. the blob keys came from an opaque GAE service so users couldn't just make them up and have them work. I think this was not intended as part of the `ValidateString()` usage, i.e. the caller just wanted type-checking.

[1]: https://github.com/dhermes/app-engine-standard/blob/19187b658bd11bac1edb0f203c22a14444fc21ff/google/appengine/api/datastore_types.py#L1267
[2]: https://github.com/dhermes/app-engine-standard/blob/19187b658bd11bac1edb0f203c22a14444fc21ff/google/appengine/api/datastore_types.py#L162